RAQUASI88/src/WIN32/retroachievements.cpp

```diff
@@ -180,18 +210,110 @@ void RA_InitMemory()
 #endif
 }
 
-void RA_OnGameClose()
+int RA_PrepareLoadNewRom(const char *file_name, int file_type)
 {
-    RA_ClearMemoryBanks();
-    RA_SetGameTitle("");
-    loaded_title[0] = '\0';
-    RA_OnLoadNewRom(NULL, 0);
+    FILE *f = fopen(file_name, "rb");
+
+    char basename[_MAX_FNAME];
+    _splitpath(file_name, NULL, NULL, basename, NULL);
+    strcpy(loading_file.name, basename);
+
+    fseek(f, 0, SEEK_END);
+    const unsigned long file_size = (unsigned long)ftell(f);
+    loading_file.data_len = file_size;
+
+    BYTE * const file_data = (BYTE *)malloc(file_size * sizeof(BYTE));
+    loading_file.data = file_data;
+    fseek(f, 0, SEEK_SET);
+    fread(file_data, sizeof(BYTE), file_size, f);
+
+    fflush(f);
+    fclose(f);
+
+    loading_file.title_id = RA_IdentifyRom(file_data, file_size);
+    loading_file.file_type = file_type;
+
+    if (loaded_title != NULL && loaded_title->data_len > 0)
+    {
+        if (loaded_title->title_id != loading_file.title_id || loaded_title->file_type != loading_file.file_type)
+        {
+            if (!RA_WarnDisableHardcore("loading a new title without ejecting all images and resetting the emulator"))
```

Comment:
this message in injected into "You cannot %s while ..." and should be simple present tense (no -ing):
```
if (!RA_WarnDisableHardcore("load a new title without ejecting all images and resetting the emulator"))
```

Reply:
Fixed.